### PR TITLE
feat(www): added breadcrumb to docs

### PR DESCRIPTION
**Pull request by gillkyle**
Co-authored-by: @mikeattara

## Description

Me and @mikeattara paired and worked this out together while pair programming 🎉 super fun!

breadcrumb

## Related Issues

Addresses #13475

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Nice!

Could we make them a bit less visually prominent? E.g. make them smaller and remove the underline?

**Comment by KyleAMathews** [comment]
Also they could use a bit of margin between them and the header. 1/2 rhythm should work.

**Comment by gillkyle** [comment]
Just made updates from the suggestions above, this is how it renders now:

image

**Comment by KyleAMathews** [comment]
Nice!

**Comment by marcysutton** [comment]
I agree the smaller font size works well! I will offer a counter-point for the link underlines that now we're only differentiating them with color, so my vote would be to keep them.

**Comment by gillkyle** [comment]
Newest commit, and underlines are back:

image

**Comment by mikeattara** [comment]
Good work @gillkyle

File: www/src/components/docs-breadcrumb.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        <Separator />
+        {activeItemParents.reverse().map(item => (
+          <React.Fragment key={item.title}>
+            <span>
+              <Link to={item.link}>{item.title}</Link>
+            </span>
+            <Separator />
+          </React.Fragment>
+        ))}
+        {activeItem.title}
```
Comment (marcysutton) [inline www/src/components/docs-breadcrumb.js]:
```suggestion
        <span aria-current="page">{activeItem.title}</span>
```
I think we should mark the current page somehow. `aria-current` is the recommended way to do that for assistive tech, and after consulting the [ARIA spec](https://www.w3.org/TR/wai-aria-1.1/#aria-current) it looks like it's allowed on any element.

Comment (gillkyle) [reply www/src/components/docs-breadcrumb.js]: I love getting this kind of feedback! Thanks for catching!

File: www/src/components/docs-breadcrumb.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+const BreadcrumbNav = ({ children, mobile = false }) => (
+  <nav
+    aria-label="breadcrumb"
+    css={{
+      fontSize: fontSizes[1],
+      display: `${mobile ? `inherit` : `none`}`,
+      [mediaQueries.md]: {
+        display: `${mobile ? `none` : `inherit`}`,
+      },
+      marginBottom: rhythm(1 / 2),
```
Comment (fk) [inline www/src/components/docs-breadcrumb.js]: There's nothing wrong with still using `rhythm`, but I think we should stick to the design tokens.

File: www/src/utils/sidebar/get-active-item-parents.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ const isParentActive = (sections, parentTitle) => {
 const getActiveItemParents = (itemList, activeItemLink, activeItemParents) => {
   if (activeItemLink.parentTitle) {
     const bar = isParentActive(itemList, activeItemLink.parentTitle)
-    activeItemParents.push(bar.title)
+    activeItemParents.push(bar)
```
Comment (fk) [inline www/src/utils/sidebar/get-active-item-parents.js]: Lol at `bar` (that was me) 🤦‍♂ … naming things is hard, but this one was lazy-stupid.

File: www/src/components/docs-breadcrumb.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import { Link } from "gatsby"
+import getActiveItem from "../utils/sidebar/get-active-item"
+import getActiveItemParents from "../utils/sidebar/get-active-item-parents"
+import { mediaQueries, space, fontSizes } from "../utils/presets"
+import { rhythm } from "../utils/typography"
+
+const Separator = ({ character = `>` }) => (
```
Comment (fk) [inline www/src/components/docs-breadcrumb.js]: I have to say I dislike using `>` (and `<`); they feel clunky, IMO they should be less wide. Maybe we can try a chevron (`react-icons/lib/md/chevron-right`)?

Comment (mikeattara) [reply www/src/components/docs-breadcrumb.js]: I think that would be cool. What's your thought? @gillkyle

Comment (marcysutton) [reply www/src/components/docs-breadcrumb.js]: That sounds good to me! I actually considered leaving feedback about the separator character, but didn't feel as strongly about it. Using a chevron would be slightly more sophisticated though.

Comment (gillkyle) [reply www/src/components/docs-breadcrumb.js]: Sounds like a great idea! Been swamped today so I could get to it later or if @mikeattara wants to take a stab at it that'd be great

Comment (gillkyle) [reply www/src/components/docs-breadcrumb.js]: Looks a lot better that way: 
image

File: www/src/components/docs-breadcrumb.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  // return a shorter version of the breadcrumb on the intro page
+  // because the docs intro page isn't generated from markdown
+  if (activeItem.title === `Introduction`) {
+    return (
+      <>
+        {/* only the breadcrumb nav of the proper viewport is displayed */}
+        <BreadcrumbNav>
+          <Link to="/">Home</Link>
+          <Separator />
+          Docs
```
Comment (marcysutton) [inline www/src/components/docs-breadcrumb.js]: This renders on the Tutorials and Contributing sections, but still says `docs`. We should fix that.

Comment (gillkyle) [reply www/src/components/docs-breadcrumb.js]: Just fixed this, now it will take the top level of the itemsList passed in and use it for the link and title displayed there which should reflect the highest item in sidebar

File: www/src/utils/sidebar/get-active-item-parents.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ const isParentActive = (sections, parentTitle) => {
 const getActiveItemParents = (itemList, activeItemLink, activeItemParents) => {
   if (activeItemLink.parentTitle) {
     const bar = isParentActive(itemList, activeItemLink.parentTitle)
-    activeItemParents.push(bar.title)
```
Comment (marcysutton) [inline www/src/utils/sidebar/get-active-item-parents.js]: This gets tricky to update the Tutorials section, which is a bit of a unique case. Currently the top-level item says "Tutorial (step-by-step)" but it's a parent to both Additional Tutorials and the upcoming Themes Tutorials, so this title doesn't make sense to be a breadcrumb parent for those. But in the context of the main tutorial where this heading shows in the sidebar, it does make sense. 

I think the best solution would be to add a `breadcrumbTitle` or something to optionally alter the breadcrumb display since even though we could split on ` (` and remove the latter part, it still isn't quite right and mucking with pluralization of a single word seems like a bad way to go. I wanted to add my comment here since I tried updating it myself to move this forward, but had to switch gears to something else for the themes launch. So we can get to this when we get to it!

Comment (gillkyle) [reply www/src/utils/sidebar/get-active-item-parents.js]: Alright I can get to this later then. My first solution actually just pulled the top level route: `/tutorial/part-one` => `tutorial` or `/docs/guides/plugins` => `docs` but maybe that wouldn't be the best solution either. Do you mean add a breadcrumbTitle to the sidebar links yaml files?

As far as comments go in those util functions, I can add some context there too.

Comment (marcysutton) [reply www/src/utils/sidebar/get-active-item-parents.js]: Ah yes, that's probably thanks to our evolving (and confusing) IA! 😅 

Yeah, I meant a value in the tutorial-links yaml file. I should have clarified! This isn't a huge rush, I just wanted to comment while it was fresh in my mind. Thanks for your work on it!

Comment (gillkyle) [reply www/src/utils/sidebar/get-active-item-parents.js]: I just pushed an update for this. It was pretty simple and works for those top level edge cases where the hierarchy might be a little weird.

**Comment by marcysutton** [comment]
Circle e2e tests never seemed to run....but I think this is ready to merge.

**Comment by lannonbr** [comment]
@marcysutton Yeah there's a config that if the PR's branch starts with docs, it will not run some of the things in CI

**Comment by marcysutton** [comment]
Merged! Woo! Thanks all for your work on this.

**Comment by mikeattara** [comment]
Happy to see this go live...

File: www/src/components/docs-breadcrumb.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    return (
+      <>
+        {/* only the breadcrumb nav of the proper viewport is displayed */}
+        <BreadcrumbNav>
+          <Link to="/">Home</Link>
+          <Separator />
+          {topLevelTitle}
+        </BreadcrumbNav>
+        <BreadcrumbNav mobile>
+          <Separator character="<" />
```
Comment (muescha) [inline www/src/components/docs-breadcrumb.js]: should it not be the ChevronLeft here also?

```suggestion
          <Separator character={<ChevronLeft />} />
```

File: www/src/data/sidebars/doc-links.yaml [hunk 8]
```diff
@@ -28,54 +29,72 @@
...
             - title: Customizing Babel.js Config
               link: /docs/babel/
+              breadcrumbTitle: Babel.js Config
             - title: Using Babel Plugin Macros
               link: /docs/babel-plugin-macros/
-            - title: Customizing Webpack Config
+              breadcrumbTitle: Babel Plugin Macros
+            - title: Adding a Custom Webpack Config
               link: /docs/add-custom-webpack-config/
+              breadcrumbTitle: webpack Config
```
Comment (muescha) [inline www/src/data/sidebars/doc-links.yaml]: uppercase?
```suggestion
              breadcrumbTitle: Webpack Config
```

File: www/src/data/sidebars/doc-links.yaml [hunk 9]
```diff
@@ -92,37 +111,50 @@
...
                 # Instead, please add links on the /docs/headless-cms/ page
                 - title: Sourcing from Contentful
                   link: /docs/sourcing-from-contentful/
+                  breadcrumbTitle: Contentful
                 - title: Sourcing from Netlify CMS
                   link: /docs/sourcing-from-netlify-cms/
+                  breadcrumbTitle: Netlify CMS
                 - title: Sourcing from WordPress
                   link: /docs/sourcing-from-wordpress/
+                  breadcrumbTitle: Wordpress
```
Comment (muescha) [inline www/src/data/sidebars/doc-links.yaml]: uppercase?
```suggestion
                  breadcrumbTitle: WordPress
```

File: www/src/data/sidebars/doc-links.yaml (hunk 8 above)
Comment (muescha) [reply www/src/data/sidebars/doc-links.yaml]: fix in #15643

File: www/src/data/sidebars/doc-links.yaml (hunk 9 above)
Comment (muescha) [reply www/src/data/sidebars/doc-links.yaml]: fix in #15643

File: www/src/components/docs-breadcrumb.js (hunk 7 above)
Comment (gillkyle) [reply www/src/components/docs-breadcrumb.js]: Yeah that's a good catch! Wanna add that into your PR in #15643?

Comment (muescha) [reply www/src/components/docs-breadcrumb.js]: ok - included in PR #15643
